internal/annotators/tpm.go

```diff
@@ -15,33 +15,37 @@ package annotators
 
 import (
 	"context"
+	"os"
+
 	"github.com/project-alvarium/alvarium-sdk-go/pkg/config"
 	"github.com/project-alvarium/alvarium-sdk-go/pkg/contracts"
 	"github.com/project-alvarium/alvarium-sdk-go/pkg/interfaces"
-	"os"
 )
 
 // Default path to a TPM 2.0 device (https://wiki.archlinux.org/title/Trusted_Platform_Module)
 const tpmPath string = "/dev/tpm0"
 
 // TpmAnnotator is used to attest whether or not the host machine has TPM capability for managing secrets
 type TpmAnnotator struct {
-	hash contracts.HashType
-	kind contracts.AnnotationType
-	sign config.SignatureInfo
+	hash  contracts.HashType
+	kind  contracts.AnnotationType
+	sign  config.SignatureInfo
+	layer contracts.LayerType
 }
 
 func NewTpmAnnotator(cfg config.SdkInfo) interfaces.Annotator {
 	a := TpmAnnotator{}
 	a.hash = cfg.Hash.Type
 	a.kind = contracts.AnnotationTPM
 	a.sign = cfg.Signature
+	a.layer = cfg.StackLayer
 	return &a
 }
 
 func (a *TpmAnnotator) Do(ctx context.Context, data []byte) (contracts.Annotation, error) {
 	key := DeriveHash(a.hash, data)
 	hostname, _ := os.Hostname()
+	tag := os.Getenv(contracts.TagEnvKey)
```

Comment:
My understanding of this line is to obtain the commit SHA of the workload in which the application is running. Assuming that's correct, note that the value will only be populated in this way for a given layer of the stack. Tag's value is contextual upon the layer, especially if the annotation is applicable to more than one layer.

What you might consider is passing the layer value into the Annotator constructor, then encapsulate the logic as to what value Tag should be assigned in the constructor. That way the decision is made in one place.

So remove `tag` as a constructor parameter
` annotation := contracts.NewAnnotation(key, a.hash, hostname, a.layer, a.kind, isSatisfied)`

And then add a `switch` inside the constructor.
```
switch layer {
case LayerType.Application:
  tag := os.Getenv(contracts.TagEnvKey)
case LayerType.CiCd, Os, Host:
  tag := "" // These are TBD. I am working on the Os / Host layers now.
}
```

Let me know if you see an issue with this or my suggestion doesn't make sense.

Reply:
I agree with your approach and have implemented the changes in the latest commit. The updated logic is now centralized in the Annotator constructor as suggested. Additionally, I've tested the changes with the [ones-demo-2021](https://github.com/michaelehab/ones-demo-2021/commit/7f9bdd3840ad848667bad30d8ea798f2c1d0888f) running in Jenkins, and everything is functioning as expected. 
![image](https://github.com/project-alvarium/alvarium-sdk-go/assets/29122581/58ce812d-5e9a-4b00-a57b-53c2ea548bcb)